tools/gyp/pylib/gyp/generator/msvs.py

```diff
@@ -3633,7 +3633,10 @@ def _AddSources2(
                     elif file_name.startswith("$("):
                         file_name = re.sub(r"^\$\([^)]+\)\\", "", file_name)
                     detail.append(["ObjectFileName", "$(IntDir)\\" + file_name])
-                grouped_sources[group].append([element, {"Include": source}] + detail)
+                element_node = [element, {"Include": source}]
+                if element == "MARMASM":
+                    element_node.append(["PreprocessedFileName", source + ".pp"])
```

Comment:
What are those `.pp` files good for?

Reply:
Somehow marmasm uses the same input file name for preprocessing output which was the RC. ".pp" file extension is one of the options for preprocessing file extension.